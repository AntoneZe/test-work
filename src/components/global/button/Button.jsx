import React from "react";
import PropTypes from "prop-types";

import "./button.less";

const Button = ({ onClick, children }) => {
  return (
    <button
      onClick={() => {
        onClick();
      }}
      className="custom-button"
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default Button;
