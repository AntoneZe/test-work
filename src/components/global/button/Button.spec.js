import React from "react";
import Button from "./Button";

const setUp = (props) => shallow(<Button {...props} />);

describe("should render Button component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render Button component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain .custom-button class", () => {
    const button = component.find(".custom-button");
    expect(button.length).toBe(1);
  });
});

describe("component callback", () => {
  let component;
  let instance;
  beforeEach(() => {
    component = setUp();
    instance = component.instance();
  });

  it("should handle Button with props", () => {
    const mockCallBack = jest.fn();

    const component = setUp({
      onClick: mockCallBack,
    });
    const btn = component.find("button");
    btn.simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it("should handle Button without props", () => {
    const mockCallBack = jest.fn();

    const component = setUp({
      onClick: undefined,
    });
    const btn = component.find("button");
    btn.simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });
});
