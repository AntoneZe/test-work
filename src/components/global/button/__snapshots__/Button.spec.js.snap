// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component should render Button component 1`] = `
<button
  className="custom-button"
  onClick={[Function]}
/>
`;
